pkg/client/apiutil/restmapper.go

```diff
@@ -162,25 +168,34 @@ func (m *mapper) addKnownGroupAndReload(groupName string, versions ...string) er
 	// If no specific versions are set by user, we will scan all available ones for the API group.
 	// This operation requires 2 requests: /api and /apis, but only once. For all subsequent calls
 	// this data will be taken from cache.
-	if len(versions) == 0 {
-		apiGroup, err := m.findAPIGroupByName(groupName)
+	//
+	// Calling this also tells us if the server supports aggregated discovery, which allows
+	// loading everything with two api calls which we assume is overall cheaper, hence we always
+	// do this once optimistically.
+	if len(versions) == 0 || !m.checkedIfSupportsAggregatedDiscovery.Load() {
```

Comment:
(I assume that aggregated discovery working once doesn't tell us anything about future discovery calls)

Would it then be better to call this something like `m.initialDiscoveryDone`?

The current name (+ the fact that we only ever set it to true) sort of implies that we check once if aggregated discovery is supported and then we'll do something with that information / it stays true for the lifetime of the binary.

Reply:
Good idea, I've renamed it to `initialDiscoveryDone`. I've also removed the `supportsAggregatedDiscovery` and changed the code to instead rely on the bool returned by `findAPIGroupByNameAndMaybeAggregatedDiscovery` to find out if aggregated discovery was done.